go/test/endtoend/transaction/twopc/fuzzer_test.go

```diff
@@ -0,0 +1,309 @@
+		ids = append(ids, (val-updateRowsVals[0][shard-1])/16)
+	}
+	return ids
+}
+
+// fuzzer runs threads that runs queries against the databases.
+// It has parameters that define the way the queries are constructed.
+type fuzzer struct {
+	threads    int
+	updateSets int
+
+	// shouldStop is an internal state variable, that tells the fuzzer
+	// whether it should stop or not.
+	shouldStop atomic.Bool
+	// wg is an internal state variable, that used to know whether the fuzzer threads are running or not.
+	wg sync.WaitGroup
+}
+
+// newFuzzer creates a new fuzzer struct.
+func newFuzzer(threads int, updateSets int) *fuzzer {
+	fz := &fuzzer{
+		threads:    threads,
+		updateSets: updateSets,
+		wg:         sync.WaitGroup{},
+	}
+	// Initially the fuzzer thread is stopped.
+	fz.shouldStop.Store(true)
+	return fz
+}
+
+// stop stops the fuzzer and waits for it to finish execution.
+func (fz *fuzzer) stop() {
+	// Mark the thread to be stopped.
+	fz.shouldStop.Store(true)
+	// Wait for the fuzzer thread to stop.
+	fz.wg.Wait()
+}
+
+// start starts running the fuzzer.
+func (fz *fuzzer) start(t *testing.T) {
+	// We mark the fuzzer thread to be running now.
+	fz.shouldStop.Store(false)
+	fz.wg.Add(fz.threads)
+	for i := 0; i < fz.threads; i++ {
+		go func() {
+			fz.runFuzzerThread(t, i)
+		}()
+	}
+}
+
+// runFuzzerThread is used to run a thread of the fuzzer.
+func (fz *fuzzer) runFuzzerThread(t *testing.T, threadId int) {
+	// Whenever we finish running this thread, we should mark the thread has stopped.
+	defer func() {
+		fz.wg.Done()
+	}()
+
+	// Create a connection to the vtgate to run transactions.
+	conn, err := mysql.Connect(context.Background(), &vtParams)
+	require.NoError(t, err)
+	defer conn.Close()
+
+	for {
+		// If fuzzer thread is marked to be stopped, then we should exit this go routine.
+		if fz.shouldStop.Load() == true {
+			return
+		}
+		// Run an atomic transaction
+		fz.generateAndExecuteTransaction(t, conn, threadId)
+	}
+
+}
+
+// initialize initializes all the variables that will be needed for running the fuzzer.
+// It also creates the rows for the `twopc_fuzzer_update` table.
+func (fz *fuzzer) initialize(t *testing.T, conn *mysql.Conn) {
+	updateRowsVals = updateRowsVals[0:1]
+	for i := 1; i < fz.updateSets; i++ {
+		updateRowsVals = append(updateRowsVals, []int{
+			updateRowsVals[0][0] + i*16,
+			updateRowsVals[0][1] + i*16,
+			updateRowsVals[0][2] + i*16,
+		})
+	}
+
+	for _, updateSet := range updateRowsVals {
+		for _, id := range updateSet {
+			_, err := conn.ExecuteFetch(fmt.Sprintf(insertIntoFuzzUpdate, id, 0), 0, false)
+			require.NoError(t, err)
+		}
+	}
+}
+
+// generateAndExecuteTransaction generates the queries of the transaction and then executes them.
+func (fz *fuzzer) generateAndExecuteTransaction(t *testing.T, conn *mysql.Conn, threadId int) {
+	// randomly generate an update set to use and the value to increment it by.
+	updateSetVal := rand.Intn(fz.updateSets)
+	incrementVal := rand.Int31()
```

Comment:
I think we should use a lower incremental value just not to go bust, which is highly unlikely but possible.

Reply:
we are not multiplying in the update, just adding. so it should be good even it it runs for 10 days.

Reply:
In the worst case it will take 50 days to go bust.